proof/verifier.go

```diff
@@ -320,6 +320,38 @@ func (p *Proof) verifyGenesisReveal() error {
 	return nil
 }
 
+// verifyGroupKeyReveal verifies that the group key reveal can be used to derive
+// the same key as the group key specified for the asset.
+func (p *Proof) verifyGroupKeyReveal() error {
+	groupKey := p.Asset.GroupKey
+	if groupKey == nil {
+		return ErrGroupKeyRequired
+	}
+
+	reveal := p.GroupKeyReveal
+	if reveal == nil {
+		return ErrGroupKeyRevealRequired
+	}
+
+	// TODO(jhb): Actually use this key and compare it.
```

Comment:
Why can't we compare/assert validity of this now?

Reply:
Depends on implementing https://github.com/lightninglabs/taproot-assets/issues/466 where the reveal is updated to the new key tweaking and signing behavior.

Reply:
Actually, we can use the current tweaking algorithm in `GroupPubKey` instead of upgrading the tweak algorithm in this PR, so that we can have this check active in this PR.